python/pyarrow/array.pxi

```diff
@@ -3609,7 +3609,7 @@ cdef class FixedShapeTensorArray(ExtensionArray):
         and the rest of the dimensions will match the permuted shape of the fixed
         shape tensor.
 
-        The conversion is zero-copy.
+        The conversion is zero-copy if data is primitive numeric and without nulls.
```

Comment:
In this case it is always zero-copy (because tensors only support numeric data without nulls, so AFAIK it would raise an error instead of being non-zero-copy)

Also, can you can the same note to the Scalar `to_numpy()` docstring?

Reply:
Indeed. Changed, Scalar as well.